perma_web/perma/tests/test_views_auth.py

```diff
@@ -1,159 +1,149 @@
+    submit_form(perma_client, 'user_management_limited_login',
+                        data = {'username': unactivated_registrar_user.email,
+                                'password': 'pass'},
+                        success_url=reverse('user_management_not_active'))
+    assert '_auth_user_id' not in perma_client.session
+
+def test_logout(perma_client, link_user):
+    """
+    Test our logout link
+    """
+
+    # Login with our client and logout with our view
+    log_in_user(perma_client, user=link_user.email, password=TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
+    perma_client.get(reverse('logout'))
+    submit_form(perma_client, 'logout')
+    assert '_auth_user_id' not in perma_client.session
+
+def test_password_change(perma_client, link_user):
+    """
+    Let's make sure we can login and change our password
+    """
+
+    log_in_user(perma_client, user=link_user.email, password=TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
+
+    perma_client.post(reverse('password_change'),
+        {'old_password':TEST_USER_PASSWORD, 'new_password1':'Changed-password1',
+        'new_password2':'Changed-password1'},
+        secure=True)
+
+    perma_client.logout()
+
+    # Try to login with our old password
+    log_in_user(perma_client, user=link_user.email, password=TEST_USER_PASSWORD)
+    assert '_auth_user_id' not in perma_client.session
+
+    perma_client.logout()
+
+    # Try to login with our new password
+    log_in_user(perma_client, user=link_user.email, password='Changed-password1')
+    assert '_auth_user_id' in perma_client.session
+
+@override_settings(AXES_FAILURE_LIMIT=2)
+def test_locked_out_after_limit(perma_client, link_user):
+    response = attempt_login(perma_client, link_user.email, 'wrongpass', expect_success=False)
+    assert response.status_code == 200
+    assert 'class="field-error"' in str(response.content)
+
+    response = attempt_login(perma_client, link_user.email, 'wrongpass', expect_success=False)
+    assert response.status_code == 403
+    assert 'Too Many Attempts' in str(response.content)
+
+    response = log_in_user(perma_client, user=link_user.email, password='Anewpass1')
+    assert response.status_code == 403
+    assert 'Too Many Attempts' in str(response.content)
+    assert '_auth_user_id' not in perma_client.session
+
+@override_settings(AXES_FAILURE_LIMIT=1)
+@override_settings(AXES_COOLOFF_TIME=datetime.timedelta(seconds=2))
+def test_lockout_expires_after_cooloff(perma_client, link_user):
+    response = attempt_login(perma_client, link_user.email, 'wrongpass', expect_success=False)
+    assert response.status_code == 403
+    assert 'Too Many Attempts' in str(response.content)
+    sleep(2)
+    log_in_user(perma_client, user=link_user.email, password=TEST_USER_PASSWORD)
+    assert '_auth_user_id' in perma_client.session
+
+
+# @override_settings(AXES_FAILURE_LIMIT=2)
+# def test_login_attempts_reset(self):
```

Comment:
Is this one commented out because it's not quite working yet? If so, lemme know if I can help :-)

Reply:
Nope, no reason! I'll put it back!